subsys/net/lib/dns/mdns_responder.c

```diff
@@ -274,6 +277,96 @@ static int send_response(struct net_context *ctx,
 	return ret;
 }
 
+static const char *qtype_to_string(int qtype)
+{
+	switch (qtype) {
+	case DNS_RR_TYPE_A: return "A";
+	case DNS_RR_TYPE_CNAME: return "CNAME";
+	case DNS_RR_TYPE_PTR: return "PTR";
+	case DNS_RR_TYPE_TXT: return "TXT";
+	case DNS_RR_TYPE_AAAA: return "AAAA";
+	case DNS_RR_TYPE_SRV: return "SRV";
+	default: return "(unsupported)";
```

Comment:
We have already `<unknown type>` string used in other parts of the net stack. So if we use that string content here, we can share the storage.